src/Components/Admin/Rooms/roomModal.js

```diff
@@ -50,24 +58,53 @@ export default function RoomModal({
         setFriendlyName(event.target.value);
     };
 
+    const handleRoomSolveTimeChange = (event) => {
+        setRoomSolveTime(event.target.value);
+    };
+
     const handleModalCloseClick = () => {
         handleModalClose();
         setRoomName("");
         setRoomDescription("");
         setFriendlyName("");
+        setRoomSolveTime("");
+        setIsPublished(false);
+        setIsPreviewable(false);
```

Comment:
I know Aaron initially added the logic here but I don't think it's needed? On each modal popup, we reset the fields in the useEffect on lines 38-47.

So in that useEffect, we can get rid of the if statement and do a ternary operator on each field. `room ? room.<field> : ""`. And we don't need to do this for edit only fields (room solve time, ispublished, and ispreviewable) since this flow is only for creating. On edit, the useEffect will reset the fields to the ones in the room's value. On create, it'll reset the exposed fields (just name, desc, friendly name) to empty strings.

I'll play around with this myself and push up a change, don't want to do delay this a few days to get this logic reworking done.

Reply:
Okay, thanks.